providers/implementations/ciphers/ciphercommon.c

```diff
@@ -435,19 +441,29 @@ int ossl_cipher_generic_block_final(void *vctx, unsigned char *out,
         return 0;
     }
 
-    if (ctx->pad && !ossl_cipher_unpadblock(ctx->buf, &ctx->bufsz, blksz)) {
-        /* ERR_raise already called */
-        return 0;
+    pad = ~constant_time_is_zero(ctx->pad);
+
+    if (pad) {
+        gp = ~constant_time_is_zero(ossl_cipher_unpadblock(ctx->buf, &ctx->bufsz, blksz));
     }
 
+    ret = constant_time_select(pad, gp, CONSTTIME_TRUE);
+
     if (outsize < ctx->bufsz) {
         ERR_raise(ERR_LIB_PROV, PROV_R_OUTPUT_BUFFER_TOO_SMALL);
         return 0;
     }
-    memcpy(out, ctx->buf, ctx->bufsz);
+
+    /* Copy ctx->buf to out in constant time */
+    for (i = 0; i < blksz; i++) {
+        i_bufsz_lt = constant_time_lt(i, ctx->bufsz);
+        buf_value = constant_time_select_s(i_bufsz_lt, ctx->buf[i], out[i]);
```

Comment:
I suggest you use `constant_time_select_8((unsigned char)(i_bufsz_lt & 0xff)), [...]`, and change buf_value to an unsigned char.